doc/display.md

```diff
@@ -33,10 +33,65 @@ i.e. `_0`, `_1`, `_2`, etc.
 The syntax does not change, but the name of the attribute is the snake case version of the trait.
 E.g. `Octal` -> `octal`, `Pointer` -> `pointer`, `UpperHex` -> `upper_hex`.
 
+# Generic data types
+
+When deriving `Display` (or other formatting trait) for a generic struct/enum, all generic type 
+arguments used during formatting are bound by respective formatting trait.
+
+E.g., for a structure `Foo` defined like this:
+```rust
+#[macro_use] extern crate derive_more;
+
+trait Trait {
+    type Type;
+}
+
+#[derive(Display)]
+#[display(fmt = "{} {} {:?} {:p}", a, b, c, d)]
+struct Foo<'a, T1, T2: Trait, T3> {
+    a: T1,
+    b: <T2 as Trait>::Type,
+    c: Vec<T3>,
+    d: &'a T1,
+}
+```
+
+The following where clauses would be generated:
+* `T1: Display + Pointer`
+* `<T2 as Trait>::Type: Debug`
+* `Bar<T3>: Display`
+
+## Custom trait bounds
+
+Sometimes you may want to specify additional trait bounds on your generic type parameters, so that they
+could be used during formatting. This can be done with a `#[display(bound = "...")]` attribute.
+
+`#[display(bound = "...")]` accepts a single string argument in a format similar to the format 
+used in angle bracket list: `T: MyTrait, U: Trait1 + Trait2`.
+
+Only type parameters defined on a struct allowed to appear in bound-string and they can only be bound
+by traits, i.e. no lifetime parameters or lifetime bounds allowed in bound-string.
+
+```rust
+#[macro_use] extern crate derive_more;
+
+trait MyTrait {
+    fn my_function(&self) -> i32;
+}
+
+#[derive(Display)]
+#[display(bound = "T: MyTrait, U: ::std::fmt::Display")]
+#[display(fmt = "{} {}", "a.my_function()", "b.to_string()")]
```

Comment:
Let's add a function call around `b.to_string`, because now this bound is not necessary.

Reply:
It's actually required. As we specify "expression" (i.e., string literal) as a fmt argument, all bets are off. No default trait bound is inferred for `U`. But I'll change this example to something more obvious.

Reply:
It's a shame that that's not working. Is it possible to keep the automatic detection of bounds and merge them with the supplied bounds? That way you wouldn't have to supply the Display bound in this case.

Reply:
Oh wait I think I misunderstood your message. You meant that this works:
```rust
#[fmt="{}", b]
```
but this doesn't?
```rust
#[fmt="{}", "b"]
```

Reply:
Exactly. Was in the process of writing a better clarification. (:

Reply:
I can change example to something like `b.to_string().len()` (as it's a more useful example) and add a clarification to documentation, that for generic parameters in cases like `#[fmt="{}", "b"]` meaningful trait bound inference is impossible.

Reply:
(I assume we're speaking about the most trivial case of just `"<field-name>"`, cause, as written in documentation, string literals in `fmt` arguments are treated as arbitrary-rust-expression-that-is-just-pasted-into-generated-implementation-as-is and we can't meaningfully infer any bounds here.)

As far as I can say, it would not be overly complex, but not absolutely trivial either. And I think such change is out of scope for this PR: implementing such bound-detection would take place in `fmt` parsing code and I've never touched that code here.

I'd also argue if implementing this feature would be reasonable at all. We already have an all-around nicer way to achieve the same result: just write `<field-name>`. It's less typing and simpler macro implementation. There are still limitations to what you can (and what you should) achieve with derive-proc-macro. (:

Reply:
@JelteF I've updated the docs.

Reply:
@ffuugoo whenever we parse code of expression or not it doesn't matter, because we don't have type information of this expression. To support such cases proc_macros are not enough and requires support of compile-time reflection in Rust.  
However, supporting tivial case like `"<field-name>"` seems feasible.

Reply:
@JelteF @ffuugoo but I still don't think this is a good choice, because now we loose the difference between `a` field and `"a"` literal.

Reply:
Let's keep it like this then